packages/eas-cli/src/credentials/ios/appstore/capabilityIdentifiers.ts

```diff
@@ -0,0 +1,103 @@
+import { BundleId, CapabilityTypeOption } from '@expo/apple-utils';
+import { JSONObject } from '@expo/json-file';
+
+import Log from '../../../log';
+import { CapabilityMapping, EXPO_NO_CAPABILITY_SYNC } from './bundleIdCapabilities';
+
+type UpdateCapabilityRequest = Parameters<BundleId['updateBundleIdCapabilityAsync']>[0];
+
+/**
+ * Sync the capability identifiers with the bundle identifier capabilities.
+ * If a capability identifier is missing, then attempt to create it.
+ * Link all of the capability identifiers at the same time after parsing the entitlements file.
+ *
+ * @param bundleId Bundle identifier object.
+ * @param entitlements JSON representation of the iOS entitlements plist
+ * @returns
+ */
+export async function syncCapabilityIdentifiersForEntitlementsAsync(
+  bundleId: BundleId,
+  entitlements: JSONObject = {}
+): Promise<{ created: string[]; linked: string[] }> {
+  if (EXPO_NO_CAPABILITY_SYNC) {
+    return { created: [], linked: [] };
+  }
+
+  const createdIds: string[] = [];
+  const linkedIds: string[] = [];
+  const CapabilityIdMapping = CapabilityMapping.filter(capability => capability.capabilityIdModel);
+
+  const updateRequest: UpdateCapabilityRequest = [];
+
+  // Iterate through the supported capabilities to build the request.
+  for (const classifier of CapabilityIdMapping) {
+    const CapabilityModel = classifier.capabilityIdModel;
+    // Skip capabilities that don't support capability IDs.
+    if (!CapabilityModel) continue;
+
+    // Skip capabilities that aren't defined in the entitlements file.
+    let capabilityIds = entitlements[classifier.entitlement] as string[];
+    if (!capabilityIds) continue;
+
+    // Assert string array matching prefix. ASC will throw if the IDs are invalid, this just saves some time.
+    if (!classifier.validateOptions(capabilityIds)) {
+      throw new Error(
+        `iOS entitlement "${classifier.entitlement}" has invalid incorrectly formatted identifier list: "${capabilityIds}".`
```

Comment:
Since we know what format to expect, it would be helpful to print what that is. That way if i provide `com.lol` as a merchant id, `eas-cli` tells me i have to add the `merchant.` prefix

Reply:
Also, if one ID is correct, and the other isn't we should say which one to correct. Right now the error message doesn't really give me anywhere to go:
```
Error: iOS entitlement "com.apple.developer.in-app-payments" has invalid incorrectly formatted identifier list: "merchant.thisoneworks,com.nonexistentid".
```